dependency/vault_pki.go

```diff
@@ -66,40 +64,68 @@ func (d *VaultPKIQuery) Fetch(clients *ClientSet, opts *QueryOptions,
 	default:
 	}
 
-	var encPEM []byte
-	var err error
-	if encPEM, err = os.ReadFile(d.filePath); err != nil || len(encPEM) == 0 {
-		// no need to write to file as it is the template dest
-		encPEM, err = d.fetchPEM(clients)
-	}
-	if err != nil {
-		return nil, nil, errors.Wrap(err, d.String())
-	}
+	needsRenewal := fmt.Errorf("needs renewal")
```

Comment:
nit: move this up?

Reply:
The 2 select statements above it are both gating code that feels better at the top. Seems clearer to me to have it closer to the use case.

Reply:
I mostly meant moving the error to top-level; since it is the always the same, there's no need to create it every time.

Reply:
This is not a general error code but more a flag to pass back to the outer logic preserving the error syntax of the call.

Maybe it could be named better to clarify this use? Idea? (that was my best shot)

Reply:
Could just be a `bool` then if you just need to flag something? Though I appreciate simplifying the return with just a single `error` instead of `error` and a `bool`...

In any case, it's a nit, so feel free to ignore.

Reply:
Thanks. I don't think a bool would improve readability at all and IMO that is sufficiently clear.

IE. I'm going to feel free. :)